### DISCUSSION TITLE
FreedCAD Mesh Issues

### DISCUSSION THREAD

**Discussion by Dono2023** [Concrete-Chrono-Development/chrono-preprocessor#12] [Q&A]
For the same cube of cross-sectional area 625 mm2, I could place particles of size 2 mm to 4 mm. On the other hand, I could not place particles of size 0.05 mm to 0.1 mm. The error is in the file attached below.

![FreedCADMeshIssues](https://github.com/Concrete-Chrono-Development/chrono-preprocessor/assets/144486857/bab33b26-e2e5-447b-9a57-076f38189968)

I appreciate your help,

Dono

**Comment by mtroemner** [comment]
Please also upload the cwPar parameter file which includes all of the selected inputs and will assist with debugging (ideally for both the working and non-working cases).

**Reply by Dono2023** [reply]
The cwPar for the working case:

[chronoWorkbench.zip](https://github.com/user-attachments/files/15568774/chronoWorkbench.zip)

The cwPar for the non-working case:

There is no cwPar file.

**Reply by mtroemner** [reply]
For the non-working case, could you please enter the values you would if you were to run the generation, then click "Write Parameters" and upload the file that is created?

**Reply by Dono2023** [reply]
Find attached the non-working case cwPar file:

[chronoWorkbench.zip](https://github.com/user-attachments/files/15595609/chronoWorkbench.zip)

**Comment by mtroemner** [comment]
Hello, it appears you have hit a practical working limit of the current preprocessor. As you have identified with your two inputs the only parameter which changes are the particle sizes and by decreasing to such a small size the current system is unable to properly function. The specific parameters you sent over for the non-working case would generate tens of millions of particles, so it is not particularly surprising. It is suggested to increase the particle diameters to a size which is possible to generate with the current or former mesh generators, or review the existing code in the repository to develop ways to improve its functionality for large scale simulations.
